### ISSUE TITLE
Disposable IAsyncCollector

### ISSUE THREAD

**Issue by Alxandr** [Azure/azure-webjobs-sdk#2264]
I'm trying to implement an `IAsyncCollector` that takes a lease on a blob while it's active, releasing the lease on `Dispose` (or `DisposeAsync`). However, neither of these are called.

#### Repro steps

Create an `IAsyncCollector`, and register it with `BindToCollector`. Then call your function, and observe that the collector is not disposed.

#### Expected behavior

The collector should be disposed.

#### Actual behavior

The collector is not disposed.

#### Known workarounds

None.

#### Related information 

Relevant packages:

```xml
<PackageReference Update="Microsoft.Azure.WebJobs" Version="3.0.9" />
<PackageReference Update="Microsoft.NET.Sdk.Functions" Version="1.0.29" />
```

**Comment by brettsam** [comment]
Have you tried doing this in `FlushAsync()`? You're correct that we don't check for IDisposable on these -- but Flush is called at the end of the function execution, so it may get you what you're looking for...

**Comment by Alxandr** [comment]
I don't want to dispose (close open connections) in `Flush`, cause I use `Flush` from the function itself (it's called multiple times).

**Comment by brettsam** [comment]
Fair enough -- it should be fairly straightforward to check if the collector implements IDisposable and then call it. 

In the meantime -- are you using this exclusively in C# as an output binding? You could get proper disposal today by creating your own service, registering it with DI as Transient and injecting it into the constructor. It can have whatever API you want and should be disposed with every function execution. Not quite the same as an output binding, but I'm not sure what your requirements are.

**Comment by Alxandr** [comment]
I wrote my own binder, cause I ran into other cases that the default `BindToCollector` couldn't deal with, so it's not really an issue for me anymore. I was quite surprised when my `IDisposable` collectors weren't disposed though, so I'd argue that this should be fixed regardless, and as you say it should be fairly easy to do.

The runtime already calls `Dispose` on binders that implement it, so it should just need to implement `Dispose` on the collector binder, and then forward the `Dispose()` down I imagine.
